ui/de-lib/src/main/java/org/iplantc/de/client/services/AppUserServiceFacade.java

```diff
@@ -82,4 +82,12 @@ void rateApp(App app,
                  AsyncCallback<AppFeedback> callback);
 
     void deleteRating(App app, AsyncCallback<AppFeedback> callback);
+
+    void getPermissions(List<App> apps, AsyncCallback<String> callback);
+
+    void shareApp(JSONObject request, AsyncCallback<String> callback);
+    
+    void unshareApp(JSONObject request, AsyncCallback<String> callback);
```

Comment:
The `shareApp` and `unshareApp` method signatures are the definition of ambiguous. Couple that with an absence of documentation, and you have no idea what is supposed to be in that `JSONObject`.

Also, nothing seems to be done with the reponses.

So, I suggest the following changes
`void shareApp(Map<String, List<Sharing>> userSharingMap, AsyncCallback<Void> callback);`
`void unshareApp(Map<String, List<Sharing>> userUnsharingMap, AsyncCallback<Void> callback);`
Then, you remove the `AppSharingPresenter.buildSharingJson()` and `AppSharingPresenter.buildUnSharingJson()` logic out of `AppSharingPresenterImpl` and into `AppUserServiceFacadeImpl`.

We've been trying to migrate that sort of json manipulation/construction out of the presenters and into the service facades anyway.